diff --git a/config/apple/codesign.py b/config/apple/codesign.py
index bb09bc493..05663c65f 100644
--- a/config/apple/codesign.py
+++ b/config/apple/codesign.py
@@ -136,6 +136,8 @@ class Bundle(object):
   def Kind(platform, extension):
     if platform in ('iphoneos', 'iphonesimulator'):
       return 'ios'
+    if platform == 'xros' or platform == 'xrsimulator':
+      return 'xros'
     if platform == 'macosx':
       if extension == '.framework':
         return 'mac_framework'
diff --git a/config/apple/mobile_config.gni b/config/apple/mobile_config.gni
index 28bba9fd1..0a33b63ae 100644
--- a/config/apple/mobile_config.gni
+++ b/config/apple/mobile_config.gni
@@ -8,6 +8,7 @@ declare_args() {
   # Configure the environment for which to build. Could be either "device",
   # "simulator" or "catalyst". Must be specified.
   target_environment = ""
+  target_os_variation = ""
 
   # Valid values: "iphoneos" (default), "tvos", "watchos".
   # Indicates the kind of iOS or iOS-based platform that is being targeted.
@@ -83,6 +84,7 @@ if (is_ios || is_watchos) {
     _target_platforms += [
       "iphoneos",
       "tvos",
+      "xros",
     ]
   } else if (is_watchos) {
     _target_platforms += [
diff --git a/config/apple/sdk_info.py b/config/apple/sdk_info.py
index 88f9cd680..298a1cd28 100755
--- a/config/apple/sdk_info.py
+++ b/config/apple/sdk_info.py
@@ -143,6 +143,8 @@ def main():
                           'macosx',
                           'watchos',
                           'watchsimulator',
+                          'xros',
+                          'xrsimulator',
                       ])
   args = parser.parse_args()
   if args.developer_dir:
diff --git a/config/clang/BUILD.gn b/config/clang/BUILD.gn
index b171ee13c..03e89943c 100644
--- a/config/clang/BUILD.gn
+++ b/config/clang/BUILD.gn
@@ -245,6 +245,14 @@ clang_lib("compiler_builtins") {
       } else {
         assert(false, "unsupported target_environment=$target_environment")
       }
+    } else if (target_platform == "xros") {
+      if (target_environment == "simulator") {
+        libname = "xrossim"
+      } else if (target_environment == "device") {
+        libname = "xros"
+      } else {
+        assert(false, "unsupported target_environment=$target_environment")
+      }
     } else {
       assert(false, "unsupported target_platform=$target_platform")
     }
diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index f5f8c05c0..ad4ab382f 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -692,22 +692,6 @@ config("compiler") {
       }
     }
 
-    # TODO(crbug.com/40283598): This causes binary size growth and potentially
-    # other problems.
-    if (default_toolchain != "//build/toolchain/cros:target") {
-      cflags += [
-        "-mllvm",
-        "-split-threshold-for-reg-with-hint=0",
-      ]
-      if (use_thin_lto && is_a_target_toolchain) {
-        if (is_win) {
-          ldflags += [ "-mllvm:-split-threshold-for-reg-with-hint=0" ]
-        } else {
-          ldflags += [ "-Wl,-mllvm,-split-threshold-for-reg-with-hint=0" ]
-        }
-      }
-    }
-
     # TODO(crbug.com/40192287): Investigate why/if this should be needed.
     if (is_win) {
       cflags += [ "/clang:-ffp-contract=off" ]
@@ -2165,6 +2149,7 @@ config("default_warnings") {
       # TODO(thakis): Push this to the (few) targets that need it,
       # instead of having a global flag.
       "-Wno-psabi",
+      "-Wno-deprecated-declarations",
     ]
 
     cflags_cc += [
diff --git a/config/ios/BUILD.gn b/config/ios/BUILD.gn
index 474b1d64b..dc3933e22 100644
--- a/config/ios/BUILD.gn
+++ b/config/ios/BUILD.gn
@@ -45,6 +45,8 @@ config("compiler") {
     triplet_os = "apple-ios"
   } else if (target_platform == "tvos") {
     triplet_os = "apple-tvos"
+  } else if (target_platform == "xros") {
+    triplet_os = "apple-xros"
   }
 
   # Set target.
diff --git a/config/ios/ios_sdk.gni b/config/ios/ios_sdk.gni
index 305f6b824..4a3a71b2e 100644
--- a/config/ios/ios_sdk.gni
+++ b/config/ios/ios_sdk.gni
@@ -91,6 +91,16 @@ if (ios_sdk_path == "") {
     } else {
       assert(false, "unsupported target_environment=$target_environment")
     }
+  } else if (target_platform == "xros") {
+    if (target_environment == "simulator") {
+      ios_sdk_name = "xrsimulator"
+      ios_sdk_platform = "XRSimulator"
+    } else if (target_environment == "device") {
+      ios_sdk_name = "xros"
+      ios_sdk_platform = "XROS"
+    } else {
+      assert(false, "unsupported environment: $target_environment")
+    }
   } else {
     assert(false, "unsupported target_platform=$target_platform")
   }
diff --git a/config/ios/rules.gni b/config/ios/rules.gni
index 6c288ee61..e7b8e0b2d 100644
--- a/config/ios/rules.gni
+++ b/config/ios/rules.gni
@@ -95,7 +95,7 @@ template("ios_info_plist") {
     if (defined(invoker.extra_substitutions)) {
       extra_substitutions += invoker.extra_substitutions
     }
-    if (target_platform == "iphoneos") {
+    if (target_platform == "iphoneos" || target_platform == "xros") {
       _build_info_plist = "//build/config/ios/BuildInfo.plist"
     } else if (target_platform == "tvos") {
       _build_info_plist = "//build/config/tvos/BuildInfo.plist"
diff --git a/config/rust.gni b/config/rust.gni
index ef45a4a9a..f7212adf7 100644
--- a/config/rust.gni
+++ b/config/rust.gni
@@ -288,6 +288,16 @@ if (is_linux || is_chromeos) {
       } else {
         assert(false, "unsupported target_environment=$target_environment")
       }
+    } else if (target_platform == "xros") {
+      if (target_environment == "simulator") {
+        rust_abi_target = "aarch64-apple-xros-sim"
+        cargo_target_abi = "sim"
+      } else if (target_environment == "device") {
+        rust_abi_target = "aarch64-apple-xros"
+        cargo_target_abi = "sim"
+      } else {
+        assert(false, "unsupported target_environment=$target_environment")
+      }
     } else {
       assert(false, "unsupported target_platform=$target_platform")
     }
@@ -308,6 +318,13 @@ if (is_linux || is_chromeos) {
       } else {
         assert(false, "unsupported target_environment=$target_environment")
       }
+    } else if (target_platform == "xros") {
+      if (target_environment == "simulator") {
+        rust_abi_target = "x86_64-apple-xros-sim"
+        cargo_target_abi = "sim"
+      } else {
+        assert(false, "unsupported target_environment=$target_environment")
+      }
     } else {
       assert(false, "unsupported target_platform=$target_platform")
     }
diff --git a/rust/known-target-triples.txt b/rust/known-target-triples.txt
index fc4780bf6..ed05a44df 100644
--- a/rust/known-target-triples.txt
+++ b/rust/known-target-triples.txt
@@ -43,3 +43,6 @@ x86_64-unknown-fuchsia
 x86_64-unknown-linux-gnu
 powerpc64le-unknown-linux-gnu
 s390x-unknown-linux-gnu
+aarch64-apple-xros
+aarch64-apple-xros-sim
+x86_64-apple-xros-sim
\ No newline at end of file
diff --git a/toolchain/apple/toolchain.gni b/toolchain/apple/toolchain.gni
index 7cac19cd0..97886c741 100644
--- a/toolchain/apple/toolchain.gni
+++ b/toolchain/apple/toolchain.gni
@@ -487,7 +487,7 @@ template("single_apple_toolchain") {
         command = "rm -f {{output}} && TOOL_VERSION=${tool_versions.filter_libtool} $python_path $script $libtool -static -D {{arflags}} -o {{output}} @$rspfile"
         description = "LIBTOOL-STATIC {{output}}"
       } else {
-        ar = "${prefix}llvm-ar"
+        ar = "../../../third_party/llvm-build/Release+Asserts/bin/llvm-ar"
         command = "\"$ar\" {{arflags}} -r -c -s -D {{output}} @$rspfile"
 
         # Remove the output file first so that ar doesn't try to modify the
