diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 4d1ac59b9..77f120dc7 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -568,10 +568,10 @@ foreach(_target_type,
         # When clang is used, some target may use clang module. This dependency
         # is required for things to work, regardless of whether the libc++
         # module is used in this target or not.
-        public_deps += [
-          "//buildtools/third_party/libc++:custom_headers",
-          "//buildtools/third_party/libc++:libcxx_headers",
-        ]
+        # public_deps += [
+        #  "//buildtools/third_party/libc++:custom_headers",
+        #  "//buildtools/third_party/libc++:libcxx_headers",
+        # ]
       }
 
       if (!defined(use_libcxx_modules)) {
@@ -597,7 +597,7 @@ foreach(_target_type,
 
       if (use_libcxx_modules) {
         # Add a dependency needed for Clang modules builds.
-        deps += [ "//buildtools/third_party/libc++:all_modules" ]
+        # deps += [ "//buildtools/third_party/libc++:all_modules" ]
       }
 
       # Consumed by the unsafe-buffers plugin during compile.
@@ -689,10 +689,10 @@ foreach(_target_type,
         # When clang is used, some target may use clang module. This dependency
         # is required for things to work, regardless of whether the libc++
         # module is used in this target or not.
-        public_deps += [
-          "//buildtools/third_party/libc++:custom_headers",
-          "//buildtools/third_party/libc++:libcxx_headers",
-        ]
+        # public_deps += [
+        #  "//buildtools/third_party/libc++:custom_headers",
+        #  "//buildtools/third_party/libc++:libcxx_headers",
+        # ]
       }
 
       if (!defined(use_libcxx_modules)) {
@@ -718,7 +718,7 @@ foreach(_target_type,
 
       if (use_libcxx_modules) {
         # Add a dependency needed for Clang modules builds.
-        deps += [ "//buildtools/third_party/libc++:all_modules" ]
+        # deps += [ "//buildtools/third_party/libc++:all_modules" ]
       }
 
       # On Android, write shared library output file to metadata. We will use
diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index be740d17b..a53df22ab 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -613,10 +613,6 @@ config("compiler") {
       ]
     }
 
-    # TODO(crbug.com/484082200): Re-enable once the test failures this causes
-    # have been solved.
-    cflags += [ "-fno-lifetime-dse" ]
-
     # TODO(hans): Remove this once Clang generates better optimized debug info
     # by default. https://crbug.com/765793
     cflags += [
@@ -1873,18 +1869,6 @@ config("sanitize_c_array_bounds") {
     cflags = [
       "-fsanitize=array-bounds",
       "-fsanitize-trap=array-bounds",
-
-      # Some code users feature detection to determine if UBSAN (or any
-      # sanitizer) is enabled, they then do expensive debug like operations. We
-      # want to suppress this behaviour since we want to keep performance costs
-      # as low as possible while having these checks.
-      "-fsanitize-ignore-for-ubsan-feature=array-bounds",
-
-      # Because we've enabled array-bounds sanitizing we also want to suppress
-      # the related warning about "unsafe-buffer-usage-in-static-sized-array",
-      # since we know that the array bounds sanitizing will catch any out-of-
-      # bounds accesses.
-      "-Wno-unsafe-buffer-usage-in-static-sized-array",
     ]
   }
 }
@@ -2214,12 +2198,6 @@ config("default_warnings") {
 
       # TODO(crbug.com/40284799): Fix and re-enable.
       "-Wno-thread-safety-reference-return",
-
-      # TODO(crbug.com/376641662): Fix and re-enable.
-      "-Wno-nontrivial-memcall",
-
-      # TODO(crbug.com/432275627): Fix and re-enable.
-      "-Wno-uninitialized-const-pointer",
     ]
 
     cflags_cc += [
diff --git a/config/ios/BUILD.gn b/config/ios/BUILD.gn
index cf3c0864f..be517c6c7 100644
--- a/config/ios/BUILD.gn
+++ b/config/ios/BUILD.gn
@@ -319,6 +319,14 @@ group("xctest") {
 _xctrunner_path =
     "$ios_sdk_platform_path/Developer/Library/Xcode/Agents/XCTRunner.app"
 
+if (target_environment == "catalyst") {
+  _xctrunner_executable_path =
+      "$_xctrunner_path/Contents/MacOS/XCTRunner"
+} else {
+  _xctrunner_executable_path =
+      "$_xctrunner_path/XCTRunner"
+}
+
 # When building with RBE, $ios_sdk_platform_path corresponds to a symlink
 # below $root_build_dir that points to the real SDK to use. Because the files
 # are below $root_build_dir, it is not possible to list them as a target input
@@ -332,15 +340,24 @@ _xctrunner_path =
 # To workaround this, add a target that pretends to create those files
 # (but does nothing). See https://crbug.com/1061487 for why this is needed.
 if (ios_use_xcode_symlinks) {
-  action("copy_xctrunner_app") {
-    testonly = true
-    script = "//build/noop.py"
+  if (target_environment == "catalyst") {
+    outputs = [
+      "$_xctrunner_path/Contents/Info.plist",
+      "$_xctrunner_path/Contents/PkgInfo",
+      "$_xctrunner_executable_path",
+    ]
+  } else {
     outputs = [
       "$_xctrunner_path/Info.plist",
       "$_xctrunner_path/PkgInfo",
-      "$_xctrunner_path/XCTRunner",
+      "$_xctrunner_executable_path",
     ]
   }
+  action("copy_xctrunner_app") {
+    testonly = true
+    script = "//build/noop.py"
+    outputs = outputs
+  }
 }
 
 # When creating the test runner for an XCUITest, the arm64e slice of the binary
@@ -349,7 +366,7 @@ if (ios_use_xcode_symlinks) {
 action("xctest_runner_without_arm64e") {
   testonly = true
   script = "//build/config/ios/strip_arm64e.py"
-  sources = [ "$_xctrunner_path/XCTRunner" ]
+  sources = [ _xctrunner_executable_path ]
   outputs = [ "$target_out_dir/XCTRunner" ]
   args = [
     "--output",
diff --git a/config/ios/rules.gni b/config/ios/rules.gni
index 8fb331121..afc83f6bb 100644
--- a/config/ios/rules.gni
+++ b/config/ios/rules.gni
@@ -1390,9 +1390,13 @@ template("ios_xcuitest_test_runner_bundle") {
     testonly = true
     script = "//build/apple/plist_util.py"
 
-    sources = [
-      "$_xctrunner_path/Info.plist",
+    if (target_environment == "catalyst") {
+      sources = ["$_xctrunner_path/Contents/Info.plist"]
+    } else {
+      sources = ["$_xctrunner_path/Info.plist"]
+    }
 
+    sources += [
       # NOTE: The XCTRunnerAddition+Info.plist must come after the Info.plist
       # because it overrides the values under "CFBundleIdentifier" and
       # "CFBundleName".
@@ -1437,7 +1441,11 @@ template("ios_xcuitest_test_runner_bundle") {
     testonly = true
     visibility = [ ":$_target_name" ]
 
-    sources = [ "$_xctrunner_path/PkgInfo" ]
+    if (target_environment == "catalyst") {
+      sources = ["$_xctrunner_path/Contents/PkgInfo"]
+    } else {
+      sources = ["$_xctrunner_path/PkgInfo"]
+    }
 
     outputs = [ "{{bundle_contents_dir}}/PkgInfo" ]
 
diff --git a/toolchain/apple/toolchain.gni b/toolchain/apple/toolchain.gni
index 804071e6b..505c849c6 100644
--- a/toolchain/apple/toolchain.gni
+++ b/toolchain/apple/toolchain.gni
@@ -214,7 +214,7 @@ template("single_apple_toolchain") {
     linker_driver_args = "-Wcrl,driver,$ld"
 
     # Specify an explicit path for the strip binary.
-    _strippath = "${prefix}llvm-strip"
+    _strippath = "${prefix}strip"
     _installnametoolpath = "${prefix}llvm-install-name-tool"
     linker_driver_args += " -Wcrl,strippath,${_strippath} -Wcrl,installnametoolpath,${_installnametoolpath}"
     _enable_dsyms = enable_dsyms
